azure-pipelines.yml

```diff
@@ -21,7 +21,20 @@ jobs:
       displayName: Install linters
     - bash: |
         set -ex
-        if [[ $BUILD_SOURCEVERSIONMESSAGE =~ \[lint\ skip\] ]]; then
+        if [[ $BUILD_REASON == "PullRequest" ]]; then
+          # By default pull requests use refs/pull/PULL_ID/merge as the source branch
+          # which has a "Merge ID into ID" as a commit message. The latest commit
+          # message is the second to last commit
```

Comment:
I assume that sometimes there is no merge commit (if the branch commits directly onto latest base)? Or does GitHub force an empty commit?

Reply:
By looking at https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=18770&view=logs&jobId=32e2e1bb-a28f-5b18-6cfc-3f01273f5609&j=32e2e1bb-a28f-5b18-6cfc-3f01273f5609&t=af038358-fc94-5f24-daa4-5c48b92b94db azure does something like this for pull request:

```bash
git init scikit-learn
cd scikit-learn
git remote add origin https://github.com/scikit-learn/scikit-learn
git fetch origin --force --tags --prune --progress  +refs/pull/17688/merge:refs/remotes/pull/17688/merge
git checkout --progress --force refs/remotes/pull/17688/merge
```

This only happens for PRs, so this bash piece only runs for `$BUILD_REASON == "PullRequest"`.

Reply:
Yes, what I'm asking is if `pull/<NUM>/merge` will always have a merge commit even if the PR was directly based on master (not behind by any commits)

Reply:
I do not think we can create a PR without no new commits:

![Screen Shot 2020-06-25 at 10 45 03 PM](https://user-images.githubusercontent.com/5402633/85815118-96d65500-b735-11ea-845d-e855f8ac0a5d.png)